components/director/internal/domain/runtime/resolver.go

```diff
@@ -710,3 +728,45 @@ func (r *Resolver) cleanupAndLogOnError(ctx context.Context, runtimeID, region s
 		log.C(ctx).Errorf("An error occurred during cleanup of self-registered runtime: %v", err)
 	}
 }
+
+func (r *Resolver) extractTenantFromSubaccountLabel(ctx context.Context, tx *persistence.PersistenceTx, value interface{}) (*model.BusinessTenantMapping, error) {
```

Comment:
Why move this to the graphQL layer? It fits in the service layer more.

Reply:
If it's moved there, CreateWithMandatoryLabels should receive a pointer to persistence.PersistenceTx in order to manipulate the transaction(possibly close it, make the API call and re-open it). But the CreateWithMandatoryLabels function is also used by Create and I am not sure how we can get the PersistenceTx there in order to pass it further. I just didn't want to change the signatures of Create and CreateWithMandatoryLabels.